docs/src/adr/002adr-types.md

```diff
@@ -25,25 +25,40 @@ when writing TLA+ specifications. This document is filling this gap.
 
 ## 1. How to write types in TLA+
 
+<a id="ts1"></a>
 ### 1.1. Type grammar (Type System 1, or TS1)
 
-**Upgrade warning.** This system will be replaced with [Type System 1.2](#ts12)
-in September of 2022.
+**Upgrade warning.** This system is replaced with [Type System 1.2](#ts12).
+In October of 2022, we will stop supporting Type System 1. For the transition
+period, pass `--features=no-rows` to Apalache, to enable Type System 1.
 
-We simply write types as strings that follow the type grammar:
+We write types as strings that follow the type grammar:
 
 ```
-T ::=   'Bool' | 'Int' | 'Str'
+T ::=   // Booleans
+      | 'Bool'
+        // integers
+      | 'Int'
+        // immutable constant strings
+      | 'Str'
+        // functions
       | T '->' T
+        // sets
       | 'Set' '(' T ')'
+        // sequences
       | 'Seq' '(' T ')'
+        // tuples
       | '<<' T ',' ...',' T '>>'
-        // warning: the following type will be removed in Type System 1.2
-      | '[' field ':' T ',' ...',' field ':' T ']'
+        // operators
       | '(' T ',' ...',' T ')' '=>' T
+        // constant types (uninterpreted types)
       | typeConst
+        // type variables
       | typeVar
+        // groups
       | '(' T ')'
```

Comment:
What are "groups"?

Reply:
well, these are just normal parentheses, which are normally use to change associativity. I will change it to something about associativity